stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/SQLDatasourceUtils.java

```diff
@@ -891,10 +891,13 @@ private static TypeKind getColumnType(int sqlType) {
         }
     }
 
-    public static String getSQLType(BType value) {
-        int tag = value.getTag();
+    public static String getSQLType(BRefType value) {
+        int tag = value.getType().getTag();
         switch (tag) {
         case TypeTags.INT_TAG:
+            if (value.value() instanceof Long) {
```

Comment:
I think this condition is always true as BInteger holds a long. So I think this check does not serve the purpose.